main/inc/lib/attendance.lib.php

```diff
@@ -2256,4 +2293,133 @@ public function exportAttendanceLogin($startDate, $endDate)
         $pdf = new PDF('A4', null, $params);
         $pdf->html_to_pdf_with_template($tableToString);
     }
+
+
+    /**
+     * Return all course for a student between dates and order by key date (Y-m-d)
+     * @param int           $student_id
+     * @param DateTime|null $startDate
+     * @param DateTime|null $endDate
+     * @param bool          $orderDesc
+     *
+     * @return array
+     */
+    public function getCoursesWithAttendance (
+        $student_id = 0,
+        DateTime $startDate = null,
+        DateTime $endDate = null,
+        $orderDesc = true
+    ) {
+
+        $courseManager = new CourseManager();
+        $attendanceLib = new Attendance();
+        $data = [];
+        // unused by now
+        /*
+         $specialCourses = $courseManager::returnSpecialCourses(
+            $student_id, false, false
+        );
+         */
+        $courses = $courseManager::returnCourses(
+            $student_id,
+            false,
+            false
+        );
+
+        /* Get course with (in_category) and without (not_category) category */
+        foreach ($courses as $coursesK => $coursesV) {
+            $coursesWithNoCategory = $coursesV;
+            $totalCoursesNoCategory = count($coursesWithNoCategory);
+            for ($i = 0; $i < $totalCoursesNoCategory; $i++) {
+                $w = $coursesWithNoCategory[$i];
+                /* Get all attendance by courses*/
+                $attenances = $attendanceLib->get_attendances_list($w['course_id']);
+                $temp = [];
+                $sheetsProccessed = [];
+                $tempDate = [];
+                foreach ($attenances as $k => $v) {
+                    $attendance_id = $v['id'];
+                    $course_id = $w['course_id'];
+                    $sheets = $attendanceLib->get_users_attendance_sheet(
+                        $attendance_id,
+                        $student_id,
+                        0,
+                        $course_id,
+                        $startDate,
+                        $endDate
+                    );
+
+                    $sheetsProccessed[] = [];
+                    foreach ($sheets as $a => $b) {
```

Comment:
Use more explanatory variables example: 
$sheet instead of $b 
$sheetKey instead of $a